lib/util/timestamp.ts

```diff
@@ -0,0 +1,75 @@
+import { DateTime } from 'luxon';
+
+export type Timestamp = string & { __timestamp: never };
+
+const timezoneOffset = new Date().getTimezoneOffset() * 60000;
+
+const millenium = 946684800000; // 2000-01-01T00:00:00.000Z
+const tomorrowOffset = 86400000; // 24 * 60 * 60 * 1000;
+
+function isValid(date: DateTime): boolean {
+  const tomorrow = DateTime.now().toMillis() + tomorrowOffset; // 24 * 60 * 60 * 1000;
```

Comment:
I would check the valid property first and early return, so we skip  some computation.